ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterAggregator.java

```diff
@@ -37,7 +36,6 @@
 
   HelixClusterAggregator(long relevantTimePeriodInMinutes) {
     relevantTimePeriodInMs = TimeUnit.MINUTES.toMillis(relevantTimePeriodInMinutes);
-    mapper.setVisibilityChecker(mapper.getVisibilityChecker().withFieldVisibility(JsonAutoDetect.Visibility.ANY));
```

Comment:
Did you set the `JsonAutoDetect` annotation because you removed this line of code? Perhaps that's a good change then, since it keeps the settings with the data class instead of with the object mapper. Still curious why it is not needed for `StatsSnapshot`, though.

Reply:
Correct. `StatsSnapshot` we have `JsonAnyGetter` and `JsonAnySetter` together with public getter/setter methods. I guess that's the reason why we don't need `JsonAutoDetect`.